lib/checkersreport.cpp

```diff
@@ -280,3 +280,89 @@ std::string CheckersReport::getReport(const std::string& criticalErrors) const
 
     return fout.str();
 }
+
+std::string CheckersReport::getXmlReport(const std::string& criticalErrors) const
+{
+    std::string ret;
+
+    ret += "    <critical-errors>";
```

Comment:
if there are no critical errors I suggest that we either don't write these at all. or we could reduce the xml to `<critical-errors/>`.